src/Core/Comparer.cs

```diff
@@ -0,0 +1,44 @@
+#region Copyright (c) 2016 Atif Aziz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#endregion
+
+namespace WebLinq
+{
+    using System;
+    using System.Collections.Generic;
+
+    public static class Comparer
```

Comment:
I renamed it to <code>ComparerExtensions</code> because we can include the same method for <code>IComparer</code> there.